src/Tickets/Commerce/Hooks.php

```diff
@@ -728,4 +731,36 @@ public function register_event_compatibility_hooks() {
 	public function include_assets_checkout_shortcode() {
 		Shortcodes\Checkout_Shortcode::enqueue_assets();
 	}
+
+	/**
+	 * Intercept the currency code option updates to set defaults according to the new selection.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $options The options being updated.
+	 *
+	 * @return array The options being updated.
+	 */
+	public function set_default_format_on_currency_change( $options  ) {
+		// bail if the currency code is not set in getting updated.
+		if ( ! isset( $_POST[ Settings::$option_currency_code ] ) ) {
+			return $options;
+		}
+
+		// get the saved currency code.
+		$saved_currency_code = tribe_get_option( Settings::$option_currency_code );
+		$new_currency_code   = sanitize_text_field( $_POST[ Settings::$option_currency_code ] );
+
+		if ( $saved_currency_code === $new_currency_code ) {
+			return $options;
+		}
+		// if they are not same then update the rest format options from default map.
+		$currency_data = Base_Commerce\Utils\Currency::get_default_currency_map();
```

Comment:
Should we add a check after this line to make sure that the currency data exists before using it?

Reply:
Updated as per suggestion. Thanks 👍